internal/core/application/service.go

```diff
@@ -1086,10 +1104,18 @@ func (s *service) FinalizeOffchainTx(
 ) (structErr errors.Error) {
 	var changes []domain.Event
 
-	offchainTx, exists := s.cache.OffchainTxs().Get(txid)
-	if !exists {
-		return errors.TX_NOT_FOUND.New("offchain tx %s not found", txid).
-			WithMetadata(errors.TxNotFoundMetadata{Txid: txid})
+	offchainTx, err := s.cache.OffchainTxs().Get(ctx, txid)
```

Comment:
The fallback masks potential bugs. We expect cache to always have the data, then a miss indicates something is wrong. Silently falling back to DB means:

  1. We won't know cache sync is broken
  2. Performance degrades (DB hit instead of cache)
  3. Could indicate data integrity issues